go/libraries/doltcore/sqle/autoincr/auto_increment_tracker.go

```diff
@@ -0,0 +1,100 @@
+package autoincr
+
+import (
+	"fmt"
+	"sync"
+)
+
+type AutoIncrementTracker interface {
+	Request(tableName string, val interface{}) (bool, error)
+	Confirm(tableName string, val interface{})
+}
+
+// AutoIncrementTracker is a global map that tracks the next auto increment value for each table in each database.
+// It is primarily used for concurrent transactions on an auto incremented table.
+func NewAutoIncrementTracker() AutoIncrementTracker {
+	return &autoIncrementTracker{
+		tables:  make(map[string]interface{}),
+		written: make(map[string]bool),
+	}
+}
+
+type autoIncrementTracker struct {
+	tables  map[string]interface{}
+	written map[string]bool
+	mu      sync.Mutex
+}
+
+var _ AutoIncrementTracker = (*autoIncrementTracker)(nil)
+
+// Request is used by an integrator to request whether the wanted auto increment key is valid for use. Requests do not
```

Comment:
This is a garbled description.

Start with what the method returns, then move on to how it's used